pumpkin-lib/src/branching/brancher.rs

```diff
@@ -68,4 +70,13 @@ pub trait Brancher {
 
     /// This method is called whenever a restart is performed.
     fn on_restart(&mut self) {}
+
+    /// This method returns whether a restart is *currently* pointless for the [`Brancher`].
+    ///
+    /// For example, if a [`Brancher`] is using a static search strategy then a restart is
+    /// pointless; however, if a [`Brancher`] is using a variable selector like [`Vsids`] which
+    /// changes throughout the search process then restarting is not pointless.
+    fn is_restart_pointless(&mut self) -> bool {
```

Comment:
Regardless of the return value of any implementation, we should document that the restart may still happen or not, depending on how different branchers are combined.